core/http-auth-aws/src/main/java/software/amazon/awssdk/http/auth/aws/internal/signer/AwsChunkedV4PayloadSigner.java

```diff
@@ -169,7 +207,49 @@ private void setupSigTrailer(ChunkedEncodedInputStream.Builder builder, RollingS
         );
     }
 
-    private void setupChecksumTrailer(ChunkedEncodedInputStream.Builder builder) {
-        // TODO(sra-identity-and-auth): Set up checksumming of chunks and add as a trailer
+    /**
+     * Add the checksum as a chunk-trailer and add it to the request's trailer header.
+     * <p>
+     * The checksum-algorithm MUST be set if this is called, otherwise it will throw.
+     */
+    private void setupChecksumTrailer(ChunkedEncodedInputStream.Builder builder, SdkHttpRequest.Builder request) {
```

Comment:
Maybe we can rename it to `setUpChecksumTrailerIfNeeded` and return if checksumAlgorithm is null? Then we can remove the checksumAlgorithm null check from the callers

Reply:
Sure, if that's an established naming pattern, I can do that